WordPress/src/main/java/org/wordpress/android/ui/sitecreation/usecases/FetchVerticalsHeaderInfoUseCase.kt

```diff
@@ -0,0 +1,62 @@
+package org.wordpress.android.ui.sitecreation.usecases
+
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.store.Store
+import org.wordpress.android.fluxc.store.VerticalStore
+import org.wordpress.android.fluxc.store.VerticalStore.VerticalErrorType
+import javax.inject.Inject
+import kotlin.coroutines.experimental.Continuation
+import kotlin.coroutines.experimental.suspendCoroutine
+
+/**
+ * Transforms EventBus event to a coroutines.
+ */
+class FetchVerticalsHeaderInfoUseCase @Inject constructor(
+    val dispatcher: Dispatcher,
+    @Suppress("unused") val verticalStore: VerticalStore
+) {
+    private var continuation: Continuation<DummyOnVerticalsHeaderInfoFetched>? = null
+
+    suspend fun fetchVerticalHeaderInfo(): DummyOnVerticalsHeaderInfoFetched {
+        if (continuation != null) {
+            throw IllegalStateException("Fetch already in progress.")
+        }
+        return suspendCoroutine { cont ->
+            continuation = cont
+            // TODO dispatcher.dispatch()
+            onVerticalHeaderInfoFetched(
+                    DummyOnVerticalsHeaderInfoFetched(
+                            DummyVerticalsHeaderInfoModel(
+                                    "What’s the focus of your business?",
+                                    "We’ll use your answer to add sections to your website.",
+                                    "eg. Landscaping, Consulting… etc"
+                            ),
+                            null
+                    )
+            )
+        }
+    }
+
+    @Subscribe(threadMode = ThreadMode.MAIN)
```

Comment:
We can use `BACKGROUND` thread for this, can't we?

Reply:
👍